safe-global/safe-wallet-web
# Refactor: load Spending limits on demand instead of global store

## What it solves

Resolves #2414

## How it solves it

* Spending limits are now not stored in a global store
* They are fetched on demand from three places:
   1. "New transaction" button – to check if the wallet is a beneficiary, if wallet isn't an owner and spending limits are enabled
   2. Spending limits settings – to display the list of existing allowances
   3. Send tokens flow – to allow sending tokens from an allowance

## How to test it

1.
* Open a Safe with spending limits
* Go to Settings and check that all the allowances are listed
* Create a new allowance
* Remove an allowance

2.
* Open a Safe with spending limits
* Connect a wallet that isn't an owner but is a beneficiary
* New Transaction should be active
* Send a token from your allowance
